core/src/main/java/io/grpc/Contexts.java

```diff
@@ -0,0 +1,239 @@
+package io.grpc;
+
+/**
+ * Utility methods for working with {@link Context}s in GRPC.
+ */
+public class Contexts {
+
+  /**
+   * Create a {@link io.grpc.ServerInterceptor} that will apply a transformation to the
+   * context associated with each {@link ServerCall} and ensure that it is
+   * {@link io.grpc.Context#current()} when dispatching calls into application callbacks.
+   * @param transformer that applies a transformation to the context.
+   * @return a {@link io.grpc.ServerInterceptor} the applies the transformation that can be bound
+   *         to a {@link io.grpc.ServerServiceDefinition}.
+   */
+  public static ServerInterceptor transformAsServerInterceptor(
+      ServerContextTransformer transformer) {
+    return new ContextAugmentingServerInterceptor(transformer);
+  }
+
+  /**
+   * Create a {@link io.grpc.ClientInterceptor} that will apply a transformation to the
+   * context associated with the {@link ClientCall} and ensure that it is
+   * {@link io.grpc.Context#current()} when dispatching calls into application callbacks.
+   * @param transformer that applies a transformation to the context.
+   * @return a {@link io.grpc.ClientInterceptor} the applies the transformation that can be bound
+   *         to a {@link io.grpc.Channel}.
+   */
+  public static ClientInterceptor transformAsClientInterceptor(
+      ClientContextTransformer transformer) {
+    return new ContextAugmentingClientInterceptor(transformer);
+  }
+
+
+  /**
+   * Transformer for server calls.
+   */
+  public static interface ServerContextTransformer {
+    /**
+     * Apply a transformation to the given {@link Context} for a received call.
+     * @param toTransform the context to transform.
+     * @param method the method that is about to be executed on the server.
+     * @param headers the headers associated with the call.
+     * @return a new context.
+     */
+    Context apply(Context toTransform, MethodDescriptor<?, ?> method, Metadata headers);
+  }
+
+  /**
+   * Transformer for client calls.
+   */
+  public static interface ClientContextTransformer {
+    /**
+     * Apply a transformation to the given {@link Context} for a dispatched call.
+     * @param toTransform the context to transform.
+     * @param method the method that is being called by the client.
+     * @param options the runtime options for the call
+     * @return a new context.
+     */
+    Context apply(Context toTransform, MethodDescriptor<?, ?> method, CallOptions options);
+  }
+
+  /**
+   * Apply a {@link ClientContextTransformer} for calls that pass through this interceptor.
+   */
+  private static class ContextAugmentingClientInterceptor implements ClientInterceptor {
```

Comment:
It's a fairly boiler plate wrapping that needs to happen. See updates use in SameProcessPropagator.